base/shell/explorer/startup.cpp

```diff
@@ -532,15 +539,28 @@ BOOL DoStartStartupItems(ITrayWindow *Tray)
             return FALSE;
     }
 
-    dwWait = WaitForSingleObject(s_hStartupMutex, INFINITE);
+    DWORD dwWait = WaitForSingleObject(s_hStartupMutex, INFINITE);
     TRACE("dwWait: 0x%08lX\n", dwWait);
     if (dwWait != WAIT_OBJECT_0)
     {
         TRACE("LastError: %ld\n", GetLastError());
 
-        DoFinishStartupItems();
+        ReleaseStartupMutex();
         return FALSE;
     }
+    return TRUE;
+}
+
+BOOL DoFinishStartupItems(VOID)
+{
+    ReleaseStartupMutex();
+    return TRUE;
+}
+
+BOOL DoStartStartupItems(ITrayWindow *Tray)
+{
+    if (!bExplorerIsShell || !InitializeStartupMutex())
```

Comment:
The `!bExplorerIsShell` may look redundant (since it's done by the `InitializeStartupMutex()`), but I may understand why you put it there: possibly to insist that all that stuff must not be done if explorer is not used as the shell :thinking: